### DISCUSSION TITLE
autozoom, autofocus

### DISCUSSION THREAD

**Discussion by nschloe** [pyvista/pyvista#2647] [Q&A]
Most of the time, I would like to position my camera such-and-such, then `zoom()` into the object to fully use the screen. I can only do this by hand now. Is there a way to set the `focal_point` to the center of the object and zoom in that everything is just in picture?

**Comment by akaszynski** [comment]
Using [set_focus](https://docs.pyvista.org/plotting/plotting.html?highlight=focus#pyvista.BasePlotter.set_focus) you can set the focus on the center of a mesh, but as for filling the screen, that's not something we've created a formal API for.

Your best bet would be to `set_focus` prior to showing the plot.  The camera will be setup in such a manner as to see the majority of the scene.

**Comment by nschloe** [comment]
> Your best bet would be to set_focus prior to showing the plot. The camera will be setup in such a manner as to see the majority of the scene.

This doesn't work too well. In my examples, it zooms in _way_ too near by default. Besides, I'd still have set the camera position which undoes the zoom anyway.
